src/object.ts

```diff
@@ -275,10 +281,10 @@ export const assign = <X extends Record<string | symbol | number, any>>(
   override: X
 ): X => {
   if (!initial || !override) return initial ?? override ?? {}
-  const merged = { ...initial }
+  const merged = cloneObject(initial)
```

Comment:
**Point of discussion**

This line changes the behavior of `assign` in two subtle ways:
1. It copies non-enumerable properties into the new object.
2. It copies the prototype into the new object, potentially calling a class constructor.

For these reasons, this *is* a breaking change, **BUT** the behavior of `assign` is so woefully underspecified that we could probably get away with publishing this in a minor version.

The question is, “what's the intuitive behavior?” One might argue it's whatever aligns with `Object.assign`, as this is basically a recursive version of that.

The best course of action is likely to implement a simpler variant of `cloneObject` that does `{ ...initial }` unless an object created with `Object.create(null)` is found, in which case, a mixture of `Object.create(null)` & `Object.assign` is performed.

Reply:
After writing that out, the answer is clear that what I described in the last paragraph is how it should work.